### PR TITLE
feat: add edit profile form

### PR DESCRIPTION
**Pull request by JohnXander**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -21,31 +21,34 @@ function App() {
...
-    const userId = getLoggedInUserId();
-    if (userId === null) {
-      return;
-    }
-    client
-      .get(`/user/${userId}`)
-      .then(res => setUser(res.data.data.user))
-      .catch(err => console.log(err));
+    const userId = getLoggedInUserId()
+    if (userId === null) { return }
```
Comment (vherus) [inline src/App.js:25]: What's the reason behind putting this on one line? It breaks the consistency of the rest of the teams

File: src/App.js [hunk 2]
```diff
@@ -21,31 +21,34 @@ function App() {
...
+
+    fetch(`${process.env.REACT_APP_API_URL}/user/${userId}`, {
```
Comment (vherus) [inline src/App.js:27]: What's the reasoning behind change the existing `client` call to a `fetch`? It's fine if there's a good reason for it, if there is you may also want to add a `.catch` to the chain to handle promise rejections

File: src/App.js [hunk 3]
```diff
@@ -21,31 +21,34 @@ function App() {
...
   const getLoggedInUserId = () => {
-    const loadedToken = localStorage.getItem("token");
-    if (loadedToken === null) {
-      return null;
-    }
-    const decoded = jwt_decode(loadedToken);
-    return decoded.userId;
-  };
+    const loadedToken = localStorage.getItem('token');
+    if (loadedToken === null) { return null }
```
Comment (vherus) [inline src/App.js:38]: Same question as above

File: src/components/profile/Profile.js [hunk 4]
```diff
@@ -1,31 +1,68 @@
...
+
+const Profile = ({ profileData, getLoggedInUserId, user, setUser }) => {
+    const { first_name, last_name, biography, github_url, cohort_id } = profileData
+    const [token] = useState(`Bearer ${localStorage.getItem(process.env.REACT_APP_USER_TOKEN)}`)
+
+    const handleSubmit = (event) => {
+        event.preventDefault()
+
+        const userId = getLoggedInUserId()
+        if (userId === null) { return }
```
Comment (vherus) [inline src/components/profile/Profile.js:17]: Same question as above

**Review by vherus — APPROVED** [review]
I'm approving this as it's good work with no real issues I can see, but please consider the comments

File: src/App.js (hunk 1 above)
Comment (JohnXander) [reply src/App.js:25]: There's no reason other than using less lines of code. 
I'll change it to suit everyone else.

**Review by vherus — APPROVED** [review]

**Review by sirpae — APPROVED** [review]

**Pull request merged**